### PR TITLE
Nn 1180 prisoner location history

### PR DESCRIPTION
**Pull request by Nimphal**
Screenshot from 2020-09-03 15-44-46

### CODE REVIEW

File: backend/controllers/prisonerProfile/prisonerCellHistory.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+} = require('../../config')
+
+module.exports = ({ elite2Api, logError, page = 0 }) => async (req, res) => {
+  const { offenderNo } = req.params
+
+  const getAgencyDetails = async cells => {
+    return Promise.all(
+      cells
+        .map(cell => cell.agencyId)
+        .filter((v, i, a) => a.indexOf(v) === i)
```
Comment (Nimphal) [inline backend/controllers/prisonerProfile/prisonerCellHistory.js:21]: Make the list unique on agency id. Prevents making multiple calls to the API for the same agency detail

File: backend/controllers/prisonerProfile/prisonerCellHistory.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if (withoutAgency.includes('RECP')) return 'Reception'
+    if (withoutAgency.includes('CSWAP')) return 'Cell swap'
+    return withoutAgency
+  }
+
+  try {
+    const { bookingId, firstName, lastName } = await elite2Api.getDetails(res.locals, offenderNo)
+    const { content: cells } = await elite2Api.getOffenderCellHistory(res.locals, bookingId, {
+      page,
+      size: 10000,
```
Comment (Nimphal) [inline backend/controllers/prisonerProfile/prisonerCellHistory.js:37]: This is temporary to return all data for now. Pagination will be properly implemented at a later stage.

**Review by jsrobertson — APPROVED** [review]

File: backend/controllers/prisonerProfile/prisonerCellHistory.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/controllers/prisonerProfile/prisonerCellHistory.js:21]: Not sure if it's better but the way we've made list unique is by doing: `Array.from(new Set(data.map(event => event.bookingId)))`

**Review by JamieDraperUK — APPROVED** [review]
LGTM 👍

**Pull request merged**